### ISSUE TITLE
Cannot search with various criteria

### ISSUE THREAD

**Issue by jumpy-jack** [OCSInventory-NG/OCSInventory-ocsreports#531]
### General informations
Operating system : Linux i686 4.12.0-0.bpo.1-686 Debian GNU/Linux 9 (stretch)

### Server installation method ( Only one choice )
- [ ] RPM Package
- [X] Install SH
- [ ] Manual installation

### Server informations
Php version : 7.0.30
Mysql / Mariadb / Percona version : Debian 9.1 version 10.1.26-MariaDB-0+deb9u1
Apache version : Apache/2.4.25 (Debian)

### OCS Inventory informations
Ocsreports version : 2.5

### Problem's description
When checking the statistics for an active package, and then checking the devices which are "notified", the following error is given:
`Warning: mysqli_fetch_object() expects parameter 1 to be mysqli_result, boolean given in /usr/share/ocsinventory-reports/ocsreports/require/function_search.php on line 617`

The URL which I am on is: 
/ocsreports/index.php?function=visu_search&prov=stat&id_pack=1532516086&stat=NOTIFIED

This is the function which is effected
```
function id_without_idgroups($list_id) {
        $sql = "select id from hardware where deviceid <> '_SYSTEMGROUP_'
                                                                                AND deviceid <> '_DOWNLOADGROUP_'
                                                                                AND id in ";
$arg = array();
        $sql = mysql2_prepare($sql, $arg, $list_id);
        $result = mysql2_query_secure($sql['SQL'], $_SESSION['OCS']["readServer"], $sql['ARG']);

        while ($item = mysqli_fetch_object($result)) {
            $res[$item->id] = $item->id;
        }
        return $res;
    }
```

I also checked the output of $_SESSION['OCS']["readServer"] and $sql which can be seen below
```
mysqli Object
(
    [affected_rows] => -1
    [client_info] => mysqlnd 5.0.12-dev - 20150407 - $Id: b5c5906d452ec590732a93b051f3827e02749b83 $
    [client_version] => 50012
    [connect_errno] => 0
    [connect_error] =>
    [errno] => 1064
    [error] => You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ')' at line 3
    [error_list] => Array
        (
            [0] => Array
                (
                    [errno] => 1064
                    [sqlstate] => 42000
                    [error] => You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ')' at line 3
                )

        )

    [field_count] => 40
    [host_info] => Localhost via UNIX socket
    [info] =>
    [insert_id] => 0
    [server_info] => 5.5.5-10.1.26-MariaDB-0+deb9u1
    [server_version] => 50505
    [stat] => Uptime: 8720300  Threads: 12  Questions: 120976992  Slow queries: 0  Opens: 3866  Flush tables: 1  Open tables: 1061  Queries per second avg: 13.873
    [sqlstate] => 00000
    [protocol_version] => 10
    [thread_id] => 49868
    [warning_count] => 0
)
Array
(
    [SQL] => select id from hardware where deviceid <> '_SYSTEMGROUP_'
                                                                                AND deviceid <> '_DOWNLOADGROUP_'
                                                                                AND id in   )
    [ARG] => Array
        (
        )
```

**Comment by jumpy-jack** [comment]
Looking into it some more, it looks like $list_id is an empty array, so could be something related to the "ms_multi_search" plugin

**Comment by jumpy-jack** [comment]
Had to roll back to previous gui version 7011

**Comment by charleneauger** [comment]
Hi @jumpy-jack ,

This bug has been fixed with PR #532.

Regards,
Charlene Auger
